homeassistant/components/ccm15/coordinator.py

```diff
@@ -0,0 +1,247 @@
+                "Data is not available for AC id %s.%s", self._host, ac_index
+            )
+            return None
+        data = self.data.devices[ac_index]
+        return data
+
+    async def async_set_hvac_mode(self, ac_index, hvac_mode: HVACMode) -> None:
+        """Set the hvac mode."""
+        _LOGGER.debug("Set Hvac[%s]='%s'", ac_index, str(hvac_mode))
+        await self.async_set_state(ac_index, "mode", CONST_STATE_CMD_MAP[hvac_mode])
+
+    async def async_set_fan_mode(self, ac_index, fan_mode: str) -> None:
+        """Set the fan mode."""
+        _LOGGER.debug("Set Fan[%s]='%s'", ac_index, fan_mode)
+        await self.async_set_state(ac_index, "fan", CONST_FAN_CMD_MAP[fan_mode])
+
+    async def async_set_temperature(self, ac_index, temp) -> None:
+        """Set the target temperature mode."""
+        _LOGGER.debug("Set Temp[%s]='%s'", ac_index, temp)
+        await self.async_set_state(ac_index, "temp", temp)
+
+
+class CCM15Climate(CoordinatorEntity[CCM15Coordinator], ClimateEntity):
+    """Climate device for CCM15 coordinator."""
+
+    def __init__(
+        self, ac_host: str, ac_index: int, coordinator: CCM15Coordinator
+    ) -> None:
+        """Create a climate device managed from a coordinator."""
+        super().__init__(coordinator)
+        self._ac_host: str = ac_host
+        self._ac_index: int = ac_index
+        self._attr_should_poll = True
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_has_entity_name = True
+        self._attr_name = f"{self._ac_index}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique id."""
+        return f"{self._ac_host}.{self._ac_index}"
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return current temperature."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("temp[%s]=%s", self._ac_index, data.temperature)
+            return data.temperature
+        return None
+
+    @property
+    def target_temperature(self) -> int | None:
+        """Return target temperature."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("set_temp[%s]=%s", self._ac_index, data.temperature_setpoint)
+            return data.temperature_setpoint
+        return None
+
+    @property
+    def target_temperature_step(self) -> int:
+        """Return target temperature step."""
+        return 1
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Return hvac mode."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            mode = data.ac_mode
+            _LOGGER.debug("hvac_mode[%s]=%s", self._ac_index, mode)
+            return CONST_CMD_STATE_MAP[mode]
+        return None
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """Return hvac modes."""
+        return [HVACMode.OFF, HVACMode.HEAT, HVACMode.COOL, HVACMode.DRY, HVACMode.AUTO]
```

Comment:
Use class variables instead for constant values.

Please also adopt it for `target_temperature_step`, `fan_modes`, `swing_modes` and `supported_features`